tests/unit_tests/cuda/propagator_cuda_kernel.cu

```diff
@@ -33,8 +33,7 @@ __global__ void propagator_test_kernel(
         return;
     }
 
-    // Set the magnetic field
-    field_type B_field(B);
+    detector_device_type::bfield_type B_field = det.get_bfield();
```

Comment:
In this case, there is no move to the device, because the field is constant and we can use the same vector value on host and device. For the pre-ACAT changes, I suggest we simply use managed memory which allows us to do the same with actual B-field maps. We can worry about moving the data properly (and not rely on GPU-side page faults) after ACAT. :cold_sweat: